lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AddressSessionService.java

```diff
@@ -294,6 +289,67 @@ private boolean validSignature(SignedJWT signedJWT, PublicKey clientPublicKey)
         }
     }
 
+    public List<CanonicalAddressWithResidency> parseAddresses(String addressBody)
+            throws AddressProcessingException {
+        List<CanonicalAddressWithResidency> addresses;
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            addresses = mapper.readValue(addressBody, List.class);
+        } catch (JsonProcessingException e) {
+            throw new AddressProcessingException(
+                    "could not parse addresses..." + e.getMessage(), e);
+        }
+
+        return addresses;
+    }
+
+    public AddressSessionItem getSession(String sessionId) {
+        return dataStore.getItem(sessionId);
+    }
+
+    public void validateSessionId(String sessionId)
+            throws SessionValidationException, SessionNotFoundException, SessionExpiredException {
+        if (sessionId == null) {
+            throw new SessionValidationException("session id is null");
+        }
+        if (sessionId.isEmpty()) {
+            throw new SessionValidationException("session id is empty");
+        }
+
+        AddressSessionItem sessionItem = dataStore.getItem(sessionId);
+        if (sessionItem == null) {
+            throw new SessionNotFoundException("session not found");
+        }
+
+        if (sessionItem.getExpiryDate() < clock.instant().getEpochSecond()) {
+            throw new SessionExpiredException("session expired");
+        }
+    }
+
+    public AddressSessionItem saveAddresses(
+            String sessionId, List<CanonicalAddressWithResidency> addresses)
+            throws SessionExpiredException, SessionValidationException, SessionNotFoundException {
+        validateSessionId(sessionId);
+
+        var sessionItem = dataStore.getItem(sessionId);
```

Comment:
in the method above we're doing the same getItem call, but follow with a null check. Is there a reason why there isn't that check here?